autogluon/task/tabular_prediction/tabular_prediction.py

```diff
@@ -393,21 +393,29 @@ def fit(train_data,
-                A `FeatureGenerator` class specifying which feature engineering protocol to follow
-                Note: The file containing your `FeatureGenerator` class must be imported into current Python session in order to use a custom class.
-            feature_generator_kwargs : dict, default={}
-                Keyword arguments to pass into the `FeatureGenerator` constructor.
+            feature_generator : `autogluon.utils.tabular.features.generators.abstract.AbstractFeatureGenerator`, default = `autogluon.utils.tabular.features.generators.auto_ml_pipeline.AutoMLPipelineFeatureGenerator()`
+                The feature generator used by AutoGluon to process the input data to the form sent to the models. This often includes automated feature generation and data cleaning.
+                It is generally recommended to keep the default feature generator unless handling an advanced use-case.
                 Valid :class:`autogluon.utils.tabular.features.auto_ml_feature_generator.AutoMLFeatureGenerator` kwargs:
-                    enable_text_ngram_features : bool, default = True
-                        If True, the vectorizer argument value is used to generate 'text_ngram' features from text features if present.
-                        Try setting this to False if you encounter memory issues running AutoGluon on text data and cannot access a machine with more memory.
-                    enable_text_special_features : bool, default = True
-                        If True, generate 'text_special' features from text features if present.
-                        Examples of 'text_special' features include the number of whitespaces and the average word length in a text feature.
-                    vectorizer : `sklearn.feature_extraction.text.CountVectorizer`, default = `CountVectorizer(min_df=30, ngram_range=(1, 3), max_features=10000, dtype=np.uint8)`
-                        Determines the count vectorizer used during feature generation if text features are detected.
-                        If your data contain text fields and you encounter memory issues running AutoGluon (and cannot access a machine with more memory), then consider reducing max_features or setting n_gram_range=(1, 2).
+                    enable_raw_features : bool, default True
+                        Enables raw feature types to be kept.
+                        This is typically any feature which is not of the types ['object', 'category', 'datetime'].
+                        Appends IdentityFeatureGenerator() to the generator group.
+                    enable_categorical_features : bool, default True
+                        Enables 'object' and 'category' feature types to be kept and processed into memory optimized category features.
```

Comment:
ditto above comment. "Enables 'object' and 'category' feature types to be kept" is not 100% clear

Reply:
improved documentation